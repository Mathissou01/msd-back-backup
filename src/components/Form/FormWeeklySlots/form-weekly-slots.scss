@use "src/styles/01-settings/settings.colors" as colors;

.c-FormWeeklySlots {
  display: flex;
  flex-direction: column;
  gap: 25px;

  &__Infos {
    display: flex;
    flex-direction: column;
    gap: 8px;
  }

  &__Title {
    color: colors.$expert-blue;
    font-weight: 700;
    font-size: 20px;
    font-family: DINPro-Bold, sans-serif;
    line-height: 25px;
  }

  &__Description {
    display: flex;
    flex-direction: column;
    line-height: 24px;
    white-space: pre-wrap;
  }

  &__Content {
    display: flex;
    flex-direction: column;
    gap: 5px;
  }

  &__Days {
    display: flex;
    flex-direction: column;
    gap: 16px;
    align-items: flex-start;
  }
}
