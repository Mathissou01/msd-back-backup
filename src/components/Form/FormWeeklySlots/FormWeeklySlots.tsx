import React from "react";
import { Controller, useFormContext } from "react-hook-form";
import { ErrorMessage } from "@hookform/error-message";
import { TSlotString, TTimeSlot, TWeeklySlots } from "../../../lib/requests";
import { IDay, weekDays } from "../../../lib/time";
import { removeNulls } from "../../../lib/utilities";
import CommonFormErrorText from "../../Common/CommonFormErrorText/CommonFormErrorText";
import TimeSlotDay, {
  ITimeSlotDayError,
  ITimeSlotDayFields,
} from "./TimeSlotDay/TimeSlotDay";
import "./form-weekly-slots.scss";

interface IFormWeeklySlotsProps {
  name: string;
  defaultValue?: TWeeklySlots;
  titleLabel?: string;
  isRequired?: boolean;
  isDisabled?: boolean;
}

export default function FormWeeklySlots({
  name,
  defaultValue,
  titleLabel,
  isRequired = false,
  isDisabled = false,
}: IFormWeeklySlotsProps) {
  /* Static Data */
  const title = titleLabel ?? "Plages horaires";
  const descriptions = {
    format: `Format attendu : 00h00`,
    example: `Exemple : 08h00 - 12h00   14h00 - 17h00`,
    instructions: `Laisser vides les jours fermés. S’il n’y a pas de pause méridienne, veuillez remplir uniquement les deux premières cases.`,
  };
  const errorMessages = {
    required: "Ce champ est obligatoire",
    validate: "Les plages horaires sont invalides",
  };

  /* Methods */
  function parseTimeSlotIntoDisplayValue(
    value?: TTimeSlot,
  ): ITimeSlotDayFields {
    const slots = value ? Object.keys(value) : [];
    let morningStart;
    let morningEnd;
    let morningSlots;
    let afternoonStart;
    let afternoonEnd;
    let afternoonSlots;
    if (value && slots.length >= 1 && slots.length <= 2) {
      const morning = slots[0] as TSlotString | undefined;
      const afternoon = slots[1] as TSlotString | undefined;
      morningStart = morning?.split("-")[0];
      morningEnd = morning?.split("-")[1];
      morningSlots = morning ? value[morning]?.fixed : undefined;
      afternoonStart = afternoon?.split("-")[0];
      afternoonEnd = afternoon?.split("-")[1];
      afternoonSlots = afternoon ? value[afternoon]?.fixed : undefined;
    }
    return {
      morningStart,
      morningEnd,
      morningSlots,
      afternoonStart,
      afternoonEnd,
      afternoonSlots,
    };
  }

  function parseTimeSlotIntoErrors(
    value?: TTimeSlot,
  ): Array<ITimeSlotDayError> {
    if (!value) return [];
    const {
      morningStart,
      morningEnd,
      morningSlots,
      afternoonStart,
      afternoonEnd,
      afternoonSlots,
    } = parseTimeSlotIntoDisplayValue(value);
    const fieldErrors: Array<ITimeSlotDayError> = [
      {
        fieldNames: ["morningStart"],
        isInvalid: (!!morningEnd || !!morningSlots) && !morningStart,
        message: "L'horaire du matin est incomplet",
      },
      {
        fieldNames: ["morningEnd"],
        isInvalid: (!!morningStart || !!morningSlots) && !morningEnd,
        message: "L'horaire du matin est incomplet",
      },
      {
        fieldNames: ["morningSlots"],
        isInvalid: (!!morningStart || !!morningEnd) && !morningSlots,
        message: "L'horaire du matin nécessite un nombre de créneaux",
      },
      {
        fieldNames: ["afternoonStart", "afternoonEnd", "afternoonSlots"],
        isInvalid:
          !morningStart &&
          !morningEnd &&
          !morningSlots &&
          (!!afternoonStart || !!afternoonStart || !!afternoonSlots),
        message:
          "Si il n'y a pas de pause méridienne, veuillez remplir uniquement l'horaire du matin",
      },
      {
        fieldNames: ["afternoonStart"],
        isInvalid: (!!afternoonEnd || !!afternoonSlots) && !afternoonStart,
        message: "L'horaire d'après-midi est incomplet",
      },
      {
        fieldNames: ["afternoonEnd"],
        isInvalid: (!!afternoonStart || !!afternoonSlots) && !afternoonEnd,
        message: "L'horaire d'après-midi est incomplet",
      },
      {
        fieldNames: ["afternoonSlots"],
        isInvalid: (!!afternoonStart || !!afternoonEnd) && !afternoonSlots,
        message: "L'horaire d'après-midi nécessite un nombre de créneaux",
      },
    ];
    return fieldErrors.filter((error) => error.isInvalid);
  }

  function validateTimeSlotDay(value: TWeeklySlots): boolean | string {
    const errors = weekDays
      .map((day) => parseTimeSlotIntoErrors(value[day.index]))
      .filter(removeNulls);
    const isInvalid = errors.length > 0;
    return !isInvalid || errorMessages.validate;
  }

  function handleChange(
    value: TWeeklySlots,
    timeSlot: TTimeSlot,
    day: IDay,
  ): TWeeklySlots {
    const newSlots = { ...value };
    newSlots[day.index] = timeSlot;
    return newSlots;
  }

  /* Local Data */
  const {
    control,
    formState: { errors },
  } = useFormContext();

  return (
    <div className="c-FormWeeklySlots">
      <div className="c-FormWeeklySlots__Infos">
        <label className="c-FormWeeklySlots__Title" htmlFor={name}>
          {title}
          {isRequired && " *"}
        </label>
        <p className="c-FormWeeklySlots__Description">
          <span>{descriptions.format}</span>
          <span>{descriptions.example}</span>
          <span>{descriptions.instructions}</span>
        </p>
      </div>
      <div className="c-FormWeeklySlots__Content">
        <div
          className="c-FormWeeklySlots__Days"
          id={name}
          data-testid="form-weekly-slots"
        >
          <Controller
            control={control}
            name={name}
            rules={{
              required: { value: isRequired, message: errorMessages.required },
              validate: validateTimeSlotDay,
            }}
            defaultValue={defaultValue}
            render={({ field: { onChange, value } }) => {
              return (
                <>
                  {weekDays.map((day, i) => (
                    <TimeSlotDay
                      key={i}
                      idPrefix={name}
                      day={day}
                      value={value?.[day.index]}
                      parseValueIntoFields={parseTimeSlotIntoDisplayValue}
                      onChange={(timeSlot) => {
                        onChange(handleChange(value, timeSlot, day));
                      }}
                      errors={parseTimeSlotIntoErrors(value?.[day.index])}
                      isDisabled={isDisabled}
                    />
                  ))}
                </>
              );
            }}
          />
        </div>
        <ErrorMessage
          errors={errors}
          name={name}
          render={({ message }: { message: string }) => (
            <CommonFormErrorText message={message} errorId={`${name}_error`} />
          )}
        />
      </div>
    </div>
  );
}
