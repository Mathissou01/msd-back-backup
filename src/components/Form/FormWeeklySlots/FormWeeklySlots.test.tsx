import { render, screen } from "@testing-library/react";
import { FormProvider, useForm } from "react-hook-form";
import React, { ReactNode } from "react";
import FormWeeklySlots from "./FormWeeklySlots";

const mock = {
  name: "form-weekly-slots",
};

const Wrapper = (props: { children: ReactNode }) => {
  const formMethods = useForm();
  return (
    <>
      <FormProvider {...formMethods}>{props.children}</FormProvider>
    </>
  );
};

describe("FormWeeklySlots", () => {
  it("renders", () => {
    const { container } = render(
      <Wrapper>
        <FormWeeklySlots name={mock.name} />
      </Wrapper>,
    );

    const inputsContainer = screen.getByTestId("form-weekly-slots");
    expect(inputsContainer).toBeInTheDocument();
    const firstInput = screen.getByTestId("form-weekly-slots-1");
    expect(firstInput).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
