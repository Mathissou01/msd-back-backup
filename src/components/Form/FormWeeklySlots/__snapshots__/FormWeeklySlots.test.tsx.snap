// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormWeeklySlots renders 1`] = `
<div>
  <div
    class="c-FormWeeklySlots"
  >
    <div
      class="c-FormWeeklySlots__Infos"
    >
      <label
        class="c-FormWeeklySlots__Title"
        for="form-weekly-slots"
      >
        Plages horaires
      </label>
      <p
        class="c-FormWeeklySlots__Description"
      >
        <span>
          Format attendu : 00h00
        </span>
        <span>
          Exemple : 08h00 - 12h00   14h00 - 17h00
        </span>
        <span>
          Laisser vides les jours fermés. S’il n’y a pas de pause méridienne, veuillez remplir uniquement les deux premières cases.
        </span>
      </p>
    </div>
    <div
      class="c-FormWeeklySlots__Content"
    >
      <div
        class="c-FormWeeklySlots__Days"
        data-testid="form-weekly-slots"
        id="form-weekly-slots"
      >
        <div
          class="c-TimeSlotDay"
        >
          <div
            class="c-TimeSlotDay__Container"
          >
            <label
              class="c-TimeSlotDay__Label"
              for="form-weekly-slots-1"
            >
              Lundi
            </label>
            <div
              class="c-TimeSlotDay__Slots"
            >
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        data-testid="form-weekly-slots-1"
                        id="form-weekly-slots-1"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-TimeSlotDay"
        >
          <div
            class="c-TimeSlotDay__Container"
          >
            <label
              class="c-TimeSlotDay__Label"
              for="form-weekly-slots-2"
            >
              Mardi
            </label>
            <div
              class="c-TimeSlotDay__Slots"
            >
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        data-testid="form-weekly-slots-2"
                        id="form-weekly-slots-2"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-TimeSlotDay"
        >
          <div
            class="c-TimeSlotDay__Container"
          >
            <label
              class="c-TimeSlotDay__Label"
              for="form-weekly-slots-3"
            >
              Mercredi
            </label>
            <div
              class="c-TimeSlotDay__Slots"
            >
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        data-testid="form-weekly-slots-3"
                        id="form-weekly-slots-3"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-TimeSlotDay"
        >
          <div
            class="c-TimeSlotDay__Container"
          >
            <label
              class="c-TimeSlotDay__Label"
              for="form-weekly-slots-4"
            >
              Jeudi
            </label>
            <div
              class="c-TimeSlotDay__Slots"
            >
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        data-testid="form-weekly-slots-4"
                        id="form-weekly-slots-4"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-TimeSlotDay"
        >
          <div
            class="c-TimeSlotDay__Container"
          >
            <label
              class="c-TimeSlotDay__Label"
              for="form-weekly-slots-5"
            >
              Vendredi
            </label>
            <div
              class="c-TimeSlotDay__Slots"
            >
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        data-testid="form-weekly-slots-5"
                        id="form-weekly-slots-5"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-TimeSlotDay"
        >
          <div
            class="c-TimeSlotDay__Container"
          >
            <label
              class="c-TimeSlotDay__Label"
              for="form-weekly-slots-6"
            >
              Samedi
            </label>
            <div
              class="c-TimeSlotDay__Slots"
            >
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        data-testid="form-weekly-slots-6"
                        id="form-weekly-slots-6"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c-TimeSlotDay"
        >
          <div
            class="c-TimeSlotDay__Container"
          >
            <label
              class="c-TimeSlotDay__Label"
              for="form-weekly-slots-0"
            >
              Dimanche
            </label>
            <div
              class="c-TimeSlotDay__Slots"
            >
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        data-testid="form-weekly-slots-0"
                        id="form-weekly-slots-0"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
              <div
                class="c-TimeSlotDay__Slot"
              >
                <div
                  class="c-TimeSlotDay__DatePickers"
                >
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="c-TimeSlotDay__Dash"
                  />
                  <div
                    class="react-datepicker-wrapper"
                  >
                    <div
                      class="react-datepicker__input-container "
                    >
                      <span
                        aria-live="polite"
                        class="react-datepicker__aria-live"
                        role="alert"
                      />
                      <input
                        class="c-TimeInput"
                        readonly=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  class="c-TimeSlotDay__SlotNumber"
                  min="1"
                  type="number"
                  value=""
                />
                <span>
                  rendez-vous
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
